src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Runtime.InteropServices;
+using System.IO;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibGpiodDriver : UnixDriver
+    {
+        private SafeChipHandle _chip;
+
+        private Dictionary<int, SafeLineHandle> _pinNumberToSafeLineHandle;
+
+        private ConcurrentDictionary<int, LibGpiodDriverEventHandler> _pinNumberToEventHandler = new ConcurrentDictionary<int, LibGpiodDriverEventHandler>();
```

Comment:
Why having two different dictionaries for the pins? Should we instead just have one and then wrap the SafeLineHandle and Event handler into one pair or an object? Is there a case where we would only add an entry in one but not on the other?

Reply:
not all pins will be used for event handling, i mean means can be opened just for read/write, pinNumberToSafeLineHandle is for all open pins, _pinNumberToEventHandler is for event handlers, its easier to handle events/delegates separately